contract/src/adjudicators/Remittance.sol

```diff
@@ -0,0 +1,134 @@
+        if (candidateData.version > previousData.version) return 1;
+        return 0;
+    }
+
+    /**
+     * @notice Identifies which participant's allocation is decreasing (the payer)
+     * @param current Current allocations
+     * @param previous Previous allocations
+     * @return payerIndex The index of the payer in the participants array
+     */
+    function identifyPayer(Allocation[] memory current, Allocation[] memory previous)
+        internal
+        pure
+        returns (uint256 payerIndex)
+    {
+        // Simple case with 2 participants
+        if (current.length != 2 || previous.length != 2) {
+            revert InvalidAllocations();
+        }
+
+        // Search for participant with decreased allocation
+        for (uint256 i = 0; i < 2; i++) {
+            if (
+                current[i].token == previous[i].token && current[i].destination == previous[i].destination
+                    && current[i].amount < previous[i].amount
+            ) {
+                return i; // Found the payer
+            }
+        }
+
+        // If no allocations decreased, revert
+        revert InvalidAllocations();
+    }
+
+    /**
+     * @notice Validates remittance state transitions
+     * @dev Ensures the payer has signed states with decreased allocations and version is incremented
+     * @param chan The channel configuration
+     * @param candidate The proposed state to be validated
+     * @param proofs Array of previous states for validation context
+     * @return valid True if the state transition is valid
+     */
+    function adjudicate(Channel calldata chan, State calldata candidate, State[] calldata proofs)
+        external
+        pure
+        override
+        returns (bool valid)
+    {
+        // Need both funding proof and previous state for validation
+        if (proofs.length < 2) {
+            return false;
+        }
+
+        // Decode the version from the candidate state
+        Voucher memory candidateData = abi.decode(candidate.data, (Voucher));
+
+        // Get the previous state (most recent proof)
+        State memory previousState = proofs[0];
+        Voucher memory previousData = abi.decode(previousState.data, (Voucher));
+
+        // Version must increase with each allocation change
+        if (candidateData.version <= previousData.version) {
+            return false;
+        }
+
+        // Compute state hashes for verification
+        bytes32 stateHash = Utils.getStateHash(chan, candidate);
+
+        // Identify the payer (participant whose allocation is decreasing)
+        uint256 payerIndex = identifyPayer(candidate.allocations, previousState.allocations);
+
+        // Payer must have signed the candidate state
+        if (
+            candidate.sigs.length <= payerIndex
+                || !Utils.verifySignature(stateHash, candidate.sigs[payerIndex], chan.participants[payerIndex])
+        ) {
+            return false;
+        }
```

Comment:
I think we can simplify and require only 1 signature, not array with signature of a correct party at some index

```suggestion
        if (
            candidate.sigs.length != 1
                || !Utils.verifySignature(stateHash, candidate.sigs[0], chan.participants[payerIndex])
        ) {
            return false;
        }
```